.github/workflows/build_and_test_workflow.yml

```diff
@@ -158,6 +158,56 @@ jobs:
       - uses: actions/upload-artifact@v3
         if: success()
         with:
-          name: ${{ matrix.name }}
+          name: ${{ matrix.suffix }}-${{ env.VERSION }}
           path: ./artifacts/target/${{ env.ARTIFACT_BUILD_NAME }}
-          retention-days: 1
\ No newline at end of file
+          retention-days: 1
+    
+  bwc-tests:
+    runs-on: ubuntu-latest
+    name: Run backwards compatibility tests
+    strategy: 
+      matrix:
+        version: [ osd-1.1.0, odfe-1.13.2, odfe-1.0.2 ]
+    steps:
+      - run: echo Running backwards compatibility tests for version ${{ matrix.version }}
+      - run: echo [NOTE] These tests will be ran using Linux x64 release builds (without security).
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version-file: "./artifacts/.nvmrc"
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Setup Yarn
+        run: |
+          npm uninstall -g yarn
+          npm i -g yarn@1.22.10
+
+      - name: Get package version
+        run: |
+          echo "VERSION=$(yarn --silent pkg-version)" >> $GITHUB_ENV
+
+      - name: Verify if OpenSearch is available for version
+        id: verify-opensearch-exists
+        run: |
+          if curl -I -L https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{ env.VERSION }}/latest/linux/x64/tar/dist/opensearch/opensearch-${{ env.VERSION }}-linux-x64.tar.gz; then
+            echo "::set-output name=version-exists::true"
+          fi
+      
+      - name: Skipping tests
+        if: steps.verify-opensearch-exists.outputs.version-exists != 'true'
+        run: echo Tests were skipped because an OpenSearch release build does not exist for this version yet!
```

Comment:
If all tests are skipped (because the format of the OpenSearch build endpoint changes, for instance), does the workflow still get marked as passing? That's fine, I guess, but it'd be a bit of a false positive.

Reply:
> If all tests are skipped (because the format of the OpenSearch build endpoint changes, for instance), does the workflow still get marked as passing? That's fine, I guess, but it'd be a bit of a false positive.

Yeah it would. But if the format changes, pretty much all the repos workflows would fail and the sanity tests pipeline will have to be updated as well. Basically, it might be a false positive but we will see failures way before hand.